src/components/avatar-audio-source.js

```diff
@@ -0,0 +1,115 @@
+const INFO_INIT_FAILED = "Failed to initialize avatar-audio-source.";
+const INFO_NO_NETWORKED_EL = "Could not find networked el.";
+const INFO_NO_OWNER = "Networked component has no owner.";
+
+// Chrome seems to require a MediaStream be attached to an AudioElement before AudioNodes work correctly
+// We don't want to do this in other browsers, particularly in Safari, which actually plays the audio despite
+// setting the volume to 0.
+const SHOULD_CREATE_SILENT_AUDIO_ELS = /chrome/i.test(navigator.userAgent);
+function createSilentAudioEl(stream) {
+  const audioEl = new Audio();
+  audioEl.setAttribute("autoplay", "autoplay");
+  audioEl.setAttribute("playsinline", "playsinline");
+  audioEl.srcObject = stream;
+  audioEl.volume = 0; // we don't actually want to hear audio from this element
+  return audioEl;
+}
+
+function getOrCreateAudioListenerForScene(scene) {
+  if (!scene.audioListener) {
+    // Would prefer not to have to do this, and am not sure if it's required,
+    // so adding a warning for now and hoping to remove this later.
+    console.warn("Scene did not have an audio listener. Creating one now.");
+    scene.audioListener = new THREE.AudioListener();
+    scene.camera && scene.camera.add(scene.audioListener);
+    scene.addEventListener("camera-set-active", function(evt) {
+      evt.detail.cameraEl.getObject3D("camera").add(scene.audioListener);
+    });
+  }
+  return scene.audioListener;
+}
+
+async function getMediaStream(el) {
+  const networkedEl = await NAF.utils.getNetworkedEntity(el).catch(e => {
+    console.error(INFO_INIT_FAILED, INFO_NO_NETWORKED_EL, e);
+  });
+  if (!networkedEl) {
+    return null;
+  }
+  const ownerId = networkedEl.components.networked.data.owner;
+  if (!ownerId) {
+    console.error(INFO_INIT_FAILED, INFO_NO_OWNER);
+    return null;
+  }
+  const stream = await NAF.connection.adapter.getMediaStream(ownerId).catch(e => {
+    console.error(INFO_INIT_FAILED, `Error getting media stream for ${ownerId}`, e);
+  });
+  if (!stream) {
+    return null;
+  }
+  return stream;
+}
+
+AFRAME.registerComponent("avatar-audio-source", {
+  schema: {
+    positional: { default: true },
+    distanceModel: {
+      default: "inverse",
+      oneOf: ["linear", "inverse", "exponential"]
+    },
+    maxDistance: { default: 10000 },
+    refDistance: { default: 1 },
+    rolloffFactor: { default: 1 }
+  },
+
+  init: async function() {
+    this.el.sceneEl.systems["hubs-systems"].audioSettingsSystem.registerAvatarAudioSource(this);
+    const stream = await getMediaStream(this.el);
```

Comment:
as I undestand it, while we can make the init async, it won't actually be treated asnyc by a-frame, that is to say, anything checking whether or not this component is initialized won't know that the `init` has actually finished or not. I don't know if that matters or not, but it seems like we should use a promise here instead of using async + await

Reply:
I think it doesn't matter because I just want to _start_ the initialization (not wait on it), but I separated an `async createAudio` function that is called from `init` and `update` to make this as clear as possible.